### ISSUE TITLE
Add TensorBoard logging for training visualization

### ISSUE THREAD

**Issue by rjwalters** [rjwalters/bucket-brigade#13]
## Goal
Integrate TensorBoard logging into the training pipeline to visualize and monitor learning progress.

## Background
Currently, training only prints periodic text updates. TensorBoard would enable:
- Real-time training curve visualization
- Hyperparameter comparison across runs
- Loss component analysis (policy loss, value loss, entropy)
- Episode statistics tracking
- Gradient flow monitoring

TensorBoard is already installed as a dependency (v2.20.0).

## Tasks
- [ ] Add TensorBoard writer to `train_simple.py`
- [ ] Log training metrics (loss, reward, episode length)
- [ ] Log PPO-specific metrics (clip fraction, KL divergence, explained variance)
- [ ] Log network statistics (gradient norms, weight distributions)
- [ ] Add hyperparameter recording
- [ ] Create example visualizations
- [ ] Document how to use TensorBoard with bucket-brigade

### Update Training Script
```python
from torch.utils.tensorboard import SummaryWriter

def train_ppo(..., run_name="default"):
    writer = SummaryWriter(f'runs/{run_name}')
    
    # Log hyperparameters
    writer.add_hparams({
        'lr': learning_rate,
        'batch_size': batch_size,
        'hidden_size': hidden_size,
        # ... etc
    }, {})
    
    # During training loop
    writer.add_scalar('train/episode_reward', avg_reward, global_step)
    writer.add_scalar('train/policy_loss', policy_loss, global_step)
    writer.add_scalar('train/value_loss', value_loss, global_step)
    writer.add_scalar('train/entropy', entropy, global_step)
    writer.add_scalar('train/clip_fraction', clip_frac, global_step)
    
    # Log gradient norms
    total_norm = sum(p.grad.norm().item()**2 for p in policy.parameters())**0.5
    writer.add_scalar('train/grad_norm', total_norm, global_step)
    
    writer.close()
```

### Metrics to Track

**Episode Metrics:**
- Mean episode reward
- Episode length
- Win rate (episodes with positive reward)
- Reward variance

**Training Metrics:**
- Policy loss
- Value loss  
- Entropy (exploration measure)
- KL divergence (policy change)
- Explained variance (value function quality)
- Clip fraction (PPO clipping activity)

**Network Metrics:**
- Gradient norms
- Weight distributions
- Learning rate (if annealed)

### Usage
```bash
# Start training with logging
uv run python scripts/train_simple.py \
    --num-steps 500000 \
    --run-name "experiment_1"

# View in TensorBoard
tensorboard --logdir runs/

# Open browser to http://localhost:6006
```

## Success Criteria
- All key metrics logged during training
- TensorBoard accessible via `tensorboard --logdir runs/`
- Multiple runs comparable in same TensorBoard session
- Documentation includes example screenshots

## Nice-to-Have
- Custom scalars for grouped metrics
- Histogram logging for action distributions
- Episode replay videos
- Profiler integration for performance analysis
- Integration with Weights & Biases as alternative

## Related
- Issue #11 (hyperparameter tuning) - TensorBoard essential for comparing runs
- Issue #10 (longer training) - monitoring is critical for long runs
